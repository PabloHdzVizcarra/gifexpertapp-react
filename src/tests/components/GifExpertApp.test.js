import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from "enzyme";
import GifExpertApp from "../../GifExpertApp";

describe('pruebas en el componente <GifExpertApp />', () => {
  
  const wrapper = shallow(<GifExpertApp />);

  test('el componente debe renderizarse correctamente', () => {
    
    expect(wrapper).toMatchSnapshot();

  })

  test('debe de mostrar una lista de categorias', () => {
    
    const categories = ['One Punch', 'Naruto'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  })
  
  
})
