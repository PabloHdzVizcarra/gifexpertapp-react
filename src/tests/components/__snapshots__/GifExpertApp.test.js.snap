// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
  </ol>
</Fragment>
`;

exports[`pruebas en el componente <GifExpertApp /> el componente debe renderizarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
